Clean checkout does not compile with "org.junit does not exist" errors

With a fresh checkout of commit 25ee80368500dd46671fcceb4b4aa6f653b59f66, Zuul is unable to build on my system (Ubuntu 14.04.1 with Oracle JDK 1.7.0_72). Builds fail with dozens of missing package errors along the lines of

```
:zuul-netflix:compileJava
warning: [options] bootstrap class path not set in conjunction with -source 1.6
/home/user/foreign-repos/zuul/zuul-netflix/src/main/java/com/netflix/zuul/scriptManager/FilterVerifier.java:22: error: package org.junit does not exist
import org.junit.Before;
                ^
/home/user/foreign-repos/zuul/zuul-netflix/src/main/java/com/netflix/zuul/scriptManager/FilterVerifier.java:23: error: package org.junit does not exist
import org.junit.Test;
                ^
/home/user/foreign-repos/zuul/zuul-netflix/src/main/java/com/netflix/zuul/scriptManager/FilterVerifier.java:26: error: package org.junit does not exist
import static org.junit.Assert.*;
                ^
```

These failures occur in both the `zuul-netflix` subdirectory and the `zuul-netflix-webapp` subdirectory (once the former is fixed). Copying the following line from `zuul-core/build.gradle` to the `build.gradle` files in the failing subdirectories allows everything to compile (though I'm not sure if this is the correct way to fix this properly):

```
provided 'junit:junit-dep:4.10'
```

--- a/zuul-netflix-webapp/build.gradle
+++ b/zuul-netflix-webapp/build.gradle
@@ -21,6 +21,7 @@ dependencies {
     compile 'com.sun.jersey:jersey-servlet:1.17.1'
 
     groovy "org.codehaus.groovy:groovy-all:2.2.2"
+    provided 'junit:junit-dep:4.10'
 }
 
 javadoc {

--- a/zuul-netflix/build.gradle
+++ b/zuul-netflix/build.gradle
@@ -19,6 +19,7 @@ dependencies {
     compile 'com.netflix.turbine:turbine-core:[0.4,)'
     
     provided 'org.powermock:powermock-api-mockito:1.5.4'
+    provided 'junit:junit-dep:4.10'
 }
 
 jar {

